src/Sentry/Ben.BlockingDetector/StaticRecursionTracker.cs

```diff
@@ -0,0 +1,11 @@
+namespace Sentry.Ben.BlockingDetector;
+
+internal class StaticRecursionTracker : IRecursionTracker
+{
+    [ThreadStatic] private static int RecursionCount;
```

Comment:
does this need to be AsyncLocal instead?

Reply:
In the original code it's a ThreadSatic. See [here](https://github.com/benaadams/Ben.BlockingDetector/blob/fce6c534dfcee6ba8cb3ef74d246211b81f5cbeb/src/Ben.BlockingDetector/BlockingMonitor.cs#L10-L11) and [here](https://github.com/benaadams/Ben.BlockingDetector/blob/fce6c534dfcee6ba8cb3ef74d246211b81f5cbeb/src/Ben.BlockingDetector/TaskBlockingListener.cs#L12-L13)... I think the intent is to report blocking only on the first occurrence. So if you have some code that does:
```
- async stuff 1
  -  async stuff 2
    - blocking stuff 3
      - blocking stuff 4
```
... you only get a blocking detection alert for blocking stuff 3.

Arguably, someone could possibly do something like this:

```
- async stuff 1
  - blocking stuff 2
    -  async stuff 3
      - blocking stuff 4
```

And there's maybe some chance blocking stuff 4 gets scheduled on the same thread as blocking stuff 2 in this case. It's a bit of an outlier though and would require a complete redesign of the blocking detector. Maybe we tackle that if it ever becomes an issue (and if we get some good engagement with this first draft of the feature).